### PR TITLE
Release 0.2.0

### PR DESCRIPTION
**Pull request by Sephsekla**
## What?

Creating the 0.2.0 release! 🥳

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+indent_style = tab
+
+[*.{js,scss}]
+indent_style = space
```
Comment (johnhooks) [inline .editorconfig:17]: Everything is currently tab based, this isn't being applied, the default for Wordpress ESLint and Prettier settings must be tabs

File: includes/restapi/fake_api.json [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	{
+		"id": 15,
+		"context": "dashboard",
+		"title": "Try this new Notification feature",
+		"source": "#WP-Notify",
+		"date": "2023-04-15T19:35:56",
+		"message": "👋 Hello from the WP Feature Notifications team! Thank you for testing out the plugin. You might want to give it a try so click on the bell icon on the right side of the adminbar.",
+		"dismissible": false,
+		"icon": {
+			"src": "https://raw.githubusercontent.com/erikyo/wp-notify/design_implementation/src/images/i.svg"
```
Comment (johnhooks) [inline includes/restapi/fake_api.json:11]: @erikyo didn't  you mention wanting to remove this icon link?

File: src/scripts/components/NoticeHubSectionHeader.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { check } from '@wordpress/icons';
+
+import { clearNotifyDrawer } from '../utils/';
```
Comment (johnhooks) [inline src/scripts/components/NoticeHubSectionHeader.js]: It's not necessary to change, but this is a remnant of the "Notify" naming.

File: src/scripts/demo/index.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+				} )
+			);
+		} );
+	}
+
+	/**
+	 *  Adding an event listener to the form with the id of `wp-notification-metabox-form` that handles "clear all notifications"
+	 */
+	const wpNotificationClearAll = document.getElementById(
+		'clear-all-wp-notify'
```
Comment (johnhooks) [inline src/scripts/demo/index.js]: More "Notify"

File: src/scripts/demo/index.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	);
+	if ( wpNotificationClearAll )
+		wpNotificationClearAll.addEventListener( 'click', () => {
+			window.wp.notify.clear( 'dashboard' );
```
Comment (johnhooks) [inline src/scripts/demo/index.js]: More "Notify"

File: src/scripts/index.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+/**
+ * Loops into contexts and adds a NoticesArea component for each one
+ */
+contexts.forEach( ( context ) => addContext( context ) );
+
+/**
+ *  exports notify store functions for further uses
+ */
+export default notify;
```
Comment (johnhooks) [inline src/scripts/index.js]: Lots of "Notify"

File: src/scripts/store/controls.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import apiFetch from '@wordpress/api-fetch';
+
+import { API_PATH } from '../constants';
+
+/**
+ * @typedef {import('./index').Notice} Notice
+ */
+/**
+ * Fetches the wp-notify rest api endpoint for the specified endpoint
```
Comment (johnhooks) [inline src/scripts/store/controls.js]: "Notify"

File: src/scripts/store/selectors.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+ *
+ * @return {Notice[]|null} the list of notices of the context
+ */
+export const getNotices = ( state, context ) => {
+	return context ? state[ context ] : null;
+};
+
+/**
+ * Adds a context to the current state.
+ * commonly it's fired when the NotifyArea is registered
```
Comment (johnhooks) [inline src/scripts/store/selectors.js]: "Notify"

File: src/scripts/utils/index.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+export const formatDate = ( date ) => {
+	return dateI18n( 'l jS F Y - h:i A', date, true );
+};
+
+/**
+ * It clears the notices in the selected context
+ *
+ * @param {string} context - The context of the notices. This is used to determine which notices to clear.
+ */
+export const clearNotifyDrawer = ( context ) => {
```
Comment (johnhooks) [inline src/scripts/utils/index.js]: "Notify"

File: src/scripts/utils/init.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+	/** If the notification context exists, render it */
+	if ( notificationContext ) {
+		/** Creates a root for Notification area, whenever is the hub or a defined area like the dashboard */
+		const componentRoot = createRoot( notificationContext );
+
+		/**
+		 * Renders the component into the specified context
+		 *
+		 * @member {HTMLElement} notifyDash - the area that will host the notifications
```
Comment (johnhooks) [inline src/scripts/utils/init.js]: "Notify"

File: storybook/.storybook/manager.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+/**
+ * External dependencies
+ */
+import {addons} from '@storybook/addons';
+
+import wpNotifyTheme from './wp-notify-theme';
+
+addons.setConfig({
+	theme: wpNotifyTheme,
```
Comment (johnhooks) [inline storybook/.storybook/manager.js]: "Notify"

File: storybook/.storybook/preview.js [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+import '../stories/assets/wp-core/buttons.css';
+import '../stories/assets/wp-core/common.css';
+import '../stories/assets/wp-core/dashboard.css';
+import '../stories/assets/wp-core/dashicons.css';
+import '../stories/assets/wp-core/edit.css';
+import '../stories/assets/wp-core/nav-menus.css';
+import '../stories/assets/wp-core/normalize.css';
+import '../stories/assets/wp-core/site-health.css';
+
+/** Wp-notify style */
```
Comment (johnhooks) [inline storybook/.storybook/preview.js]: "Notify"

File: storybook/.storybook/wp-notify-theme.js [hunk 13]
```diff
@@ -0,0 +1,10 @@
+import {create} from '@storybook/theming';
+import Logo from '../stories/assets/logo.svg';
+
+export default create({
+	base: 'light',
+	brandTitle: 'WP-Notify storybook',
```
Comment (johnhooks) [inline storybook/.storybook/wp-notify-theme.js]: "Notify"

File: storybook/stories/Dash-multiple.stories.jsx [hunk 14]
```diff
@@ -0,0 +1,50 @@
+/** Wp-notify style */
```
Comment (johnhooks) [inline storybook/stories/Dash-multiple.stories.jsx]: "Notify"

File: storybook/stories/Dash-single.stories.jsx [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
+const Template = ( args ) => (
+	<div
+		id="wpbody"
+		style={ {
+			fontFamily:
+				'-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif',
+		} }
+	>
+		<div id="wp-notify-dashboard-notices" className="wrap">
```
Comment (johnhooks) [inline storybook/stories/Dash-single.stories.jsx]: "Notify"

File: storybook/stories/Dash-single.stories.jsx [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+		</div>
+	</div>
+);
+
+export const single = Template.bind( {} );
+// More on args: https://storybook.js.org/docs/react/writing-stories/args
+single.args = {
+	id: 15,
+	title: 'Try this new Notification feature',
+	source: '#WP-Notify',
```
Comment (johnhooks) [inline storybook/stories/Dash-single.stories.jsx]: "Notify"

File: storybook/stories/Hub-single.stories.jsx [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+		</div>
+	);
+};
+
+export const single = Template.bind( {} );
+// More on args: https://storybook.js.org/docs/react/writing-stories/args
+single.args = {
+	id: 15,
+	title: 'Try this new Notification feature',
+	source: '#WP-Notify',
```
Comment (johnhooks) [inline storybook/stories/Hub-single.stories.jsx]: "Notify"

File: storybook/stories/docs/contributors/develop.md [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+
+```bash
+vendor/bin/phpunit
+```
+
+## Developing with wp-env
+
+The [wp-env package](https://developer.wordpress.org/block-editor/packages/packages-env/) was developed with the Gutenberg project as a quick way to create a standard WordPress environment using Docker. It is also published as the `@wordpress/env` npm package.
+
+You can use it for contributing to the WP Notify project, but you need to install it on your computer first. read the [prerequisites](https://developer.wordpress.org/block-editor/packages/packages-env/#prerequisites) and the [install as a global package](https://developer.wordpress.org/block-editor/packages/packages-env/#installation-as-a-global-package) from its manual.
```
Comment (johnhooks) [inline storybook/stories/docs/contributors/develop.md]: "Notify"

File: includes/restapi/fake_api.json [hunk 19]
```diff
@@ -0,0 +1,171 @@
...
+		"action": {
+			"acceptMessage": "Test",
+			"acceptLink": "#"
+		}
+	},
+	{
+		"id": 13,
+		"context": "adminbar",
+		"title": "Message variant #2",
+		"source": "#WP-Notify",
```
Comment (johnhooks) [inline includes/restapi/fake_api.json]: "Notify"

**Review by johnhooks — CHANGES_REQUESTED** [review]
Should we remove all these instances of the word notify?

File: .editorconfig (hunk 1 above)
Comment (Sephsekla) [reply .editorconfig:17]: I can't remember where this came from but I'm personally in favour of tabs for everything, we could maybe change it.

**Comment by Sephsekla** [comment]
Good spot on all of these!

I suggest we open a PR into develop for those notify references, and maybe another one for the editorconfig.

File: .editorconfig (hunk 1 above)
Comment (johnhooks) [reply .editorconfig:17]: I agree tabs all day

**Review by johnhooks — DISMISSED** [review]
Let's ship this thing

**Review by erikyo — APPROVED** [review]
LGTM!

**Review by johnhooks — APPROVED** [review]

**Pull request merged**